src/antsibull/build_changelog.py

```diff
@@ -69,64 +112,117 @@ def optimize_release_entry(entry: ChangelogGeneratorEntry) -> ChangelogGenerator
+
+
+def _add_table_line(
+    lines: list[str],
+    column_widths: list[int],
+    start_sep: str,
+    mid_sep: str,
+    end_sep: str,
+    corner: str,
+):
+    parts = [corner]
     for w in column_widths:
-        parts.append(sep * (w + 2) + "+")
-    builder.add_raw_rst("".join(parts))
+        parts.append(start_sep)
+        parts.append(mid_sep * w)
+        parts.append(end_sep)
+        parts.append(corner)
+    lines.append("".join(parts))
 
 
-def render_rst_table(
-    builder: RstBuilder, headings: list[str], cells: list[list[str]]
-) -> None:
-    # Determine column widths
+def compute_column_widths(headings: list[str], cells: list[list[str]]) -> list[int]:
     column_widths: list[int] = []
     for row in [headings] + cells:
         while len(row) > len(column_widths):
             column_widths.append(0)
         for i, value in enumerate(row):
             for line in value.splitlines():
                 column_widths[i] = max(column_widths[i], len(line))
+    return column_widths
+
+
+def render_rst_table(headings: list[str], cells: list[list[str]]) -> str:
+    column_widths = compute_column_widths(headings, cells)
+    lines: list[str] = []
+    _add_table_line(lines, column_widths, "-", "-", "-", "+")
+    _add_table_row(lines, column_widths, headings, "|")
+    _add_table_line(lines, column_widths, "=", "=", "=", "+")
+    for row in cells:
+        _add_table_row(lines, column_widths, row, "|")
+        _add_table_line(lines, column_widths, "-", "-", "-", "+")
+    return "\n".join(lines)
+
 
-    # Add rows
-    _add_rst_table_line(builder, column_widths, "-")
-    _add_rst_table_row(builder, column_widths, headings)
-    _add_rst_table_line(builder, column_widths, "=")
+def render_md_table(headings: list[str], cells: list[list[str]]) -> str:
+    column_widths = compute_column_widths(headings, cells)
+    lines: list[str] = []
+    _add_table_row(lines, column_widths, headings, "|")
+    _add_table_line(lines, column_widths, " ", "-", " ", "|")
     for row in cells:
-        _add_rst_table_row(builder, column_widths, row)
-        _add_rst_table_line(builder, column_widths, "-")
+        _add_table_row(lines, column_widths, row, "|")
+    return "\n".join(lines)
+
+
+def format_link(title: str, url: str, text_format: TextFormat) -> str:
+    if text_format == TextFormat.RESTRUCTURED_TEXT:
+        return f"`{title} <{url}>`_"
+    if text_format == TextFormat.MARKDOWN:
+        return f"`[{title}]({url})"
+    raise ValueError(f"Unknown format {format}")
+
+
+def render_table(
+    renderer: AbstractRenderer,
+    headings: list[str],
+    cells: list[list[str]],
+    text_format: TextFormat,
+) -> None:
+    if text_format == TextFormat.RESTRUCTURED_TEXT:
+        renderer.add_text(render_rst_table(headings, cells), text_format=text_format)
+        return
+    if text_format == TextFormat.MARKDOWN:
+        renderer.add_text(render_md_table(headings, cells), text_format=text_format)
+        return
```

Comment:
Something like

``` python
render_table = {TextFormat.RESTRUCTURED_TEXT: render_rst_table, TextFormat.MARKDOWN: render_md_table}[text_format]
renderer.add_txt(render_table(headings, cells), text_format=text_format)
```

(with error handling) might be a better solution.

Reply:
c7aa28e90c67b0541fa130ab265248c93b707a87